From a6ccd06741bc0c9b51261b87614fd7c60416a64e Mon Sep 17 00:00:00 2001
From: Aaro Koskinen <aaro.koskinen@iki.fi>
Date: Thu, 25 Aug 2016 18:24:53 -0700
Subject: [PATCH 4/5] v4.8-rc: gpio-leds broken on OCTEON

Hi,

On Wed, Aug 24, 2016 at 11:42:00AM -0500, Steven J. Hill wrote:
> It is actually two patches that cause the breakage. The other is:
>
>    commit e55aeb6ba4e8cc3549bff1e75ea1d029324bce21
>    of/irq: Mark interrupt controllers as populated before initialisation
>
> I needed to revert both of these in order to get MMC working on our 71xx and
> 78xx boards. For our MMC, I got error messages from the MMC core of "Invalid
> POWER GPIO" until I applied the second patch. I will have a fix worthy of
> upstreaming today which will be posted today.

The below change works for me...

Signed-off-by: Stephen Arnold <sarnold@vctlabs.com>
---
 arch/mips/cavium-octeon/octeon-irq.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/cavium-octeon/octeon-irq.c b/arch/mips/cavium-octeon/octeon-irq.c
index 5a9b87b..5fd57c2 100644
--- a/arch/mips/cavium-octeon/octeon-irq.c
+++ b/arch/mips/cavium-octeon/octeon-irq.c
@@ -1618,6 +1618,7 @@ static int __init octeon_irq_init_gpio(
 		pr_warn("Cannot allocate memory for GPIO irq_domain.\n");
 		return -ENOMEM;
 	}
+	of_node_clear_flag(gpio_node, OF_POPULATED);
 
 	return 0;
 }
-- 
2.9.0

